www/include/options/accessLists/resourcesACL/DB-Func.php

```diff
@@ -346,38 +348,43 @@ function insertLCA()
 /**
  *
  * Update resource ACL in DB
- * @param $acl_id
+ * @param int|null $aclId
  */
-function updateLCA($acl_id = null)
+function updateLCA(?int $aclId = null): void
 {
     global $form, $pearDB;
 
-    if (!$acl_id) {
+    if (!$aclId) {
         return;
     }
 
-    $ret = array();
+    $ret = [];
     $ret = $form->getSubmitValues();
 
-    $rq = "UPDATE `acl_resources` ";
-    $rq .= "SET acl_res_name = '" . $pearDB->escape($ret["acl_res_name"]) . "', " .
-        "acl_res_alias = '" . $pearDB->escape($ret["acl_res_alias"]) . "', " .
-        "all_hosts = '" . (isset($ret["all_hosts"]["all_hosts"])
-            ? $pearDB->escape($ret["all_hosts"]["all_hosts"])
-            : 0) . "', " .
-        "all_hostgroups = '" . (isset($ret["all_hostgroups"]["all_hostgroups"])
-            ? $pearDB->escape($ret["all_hostgroups"]["all_hostgroups"])
-            : 0) . "', " .
-        "all_servicegroups = '" . (isset($ret["all_servicegroups"]["all_servicegroups"])
-            ? $pearDB->escape($ret["all_servicegroups"]["all_servicegroups"])
-            : 0) . "', " .
-        "acl_res_activate = '" . $pearDB->escape($ret["acl_res_activate"]["acl_res_activate"]) . "', " .
-        "acl_res_comment = '" . $pearDB->escape(!isset($ret["acl_res_comment"])
-            ? ""
-            : $ret["acl_res_comment"]) . "', " .
-        "changed = '1' " .
-        "WHERE acl_res_id = '" . $acl_id . "'";
-    $dbResult = $pearDB->query($rq);
+    $resourceValues = sanitizeResourceParameters($ret);
+
+    $statement = $pearDB->prepare(
+        "UPDATE `acl_resources`
+        SET acl_res_name = :aclResourceName,
+        acl_res_alias = :aclResourceAlias,
+        all_hosts = :allHosts,
+        all_hostgroups = :allHostGroups,
+        all_servicegroups = :allServiceGroups,
+        acl_res_activate = :aclResourceActivate,
+        acl_res_comment = :aclResourceComment,
+        changed = 1
+        WHERE acl_res_id = :aclId
+        "
+    );
+    $statement->bindValue(':aclResourceName', $resourceValues['acl_res_name'], \PDO::PARAM_STR);
+    $statement->bindValue(':aclResourceAlias', $resourceValues['acl_res_alias'], \PDO::PARAM_STR);
```

Comment:
Be careful, a boolean value equal to false (filter_var) will create an empty value with the string format.
Is this the desired behaviour?

Reply:
as discuss with @kduret , acl_res_name is a mandatory field, so if Resource Name is an empty string, an exception will be thrown (13adc79), for alias and comment, an empty string will be inserted as they aren't mandatory fields.